stackslib/src/chainstate/stacks/boot/pox-4.clar

```diff
@@ -1335,4 +1373,10 @@
     (begin
         (ok (map-set aggregate-public-keys reward-cycle aggregate-public-key))
     )
-)
\ No newline at end of file
+)
+
+;; Check if a provided signer key is valid. For now it only asserts length.
+;; *New in Stacks 3.0*
+(define-read-only (is-signer-key-valid (signer-key (buff 33)))
```

Comment:
There's no realistic way to ensure that the key is valid.  The caller may put a well-formed public key in that has no corresponding private key, for example.  I'd just drop this entirely.

Reply:
It is mainly to prevent user error or faulty clients, just like how we have `check-pox-addr-version` and `check-pox-addr-hashbytes`. In `is-signer-key-valid` we can check if the key length is `33` and the prefix is `0x02` or `0x03`. I can rename it to `check-signer-key` to prevent misconceptions. Still want me to remove?

Reply:
This function was renamed `insert-signer-key` and now does two things:

1. Check that the provided signer key is 33 bytes.
2. Assert-insert the signer key into a map to prevent re-use of the key.

Building this logic out further to support aggregate key calculation will be done in the next PR.